util.go

```diff
@@ -0,0 +1,80 @@
+package files
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"sort"
+)
+
+// ToFile is an alias for n.(File). If the file isn't a regular file, nil value
+// will be returned
+func ToFile(n Node) File {
+	f, _ := n.(File)
+	return f
+}
+
+// ToDir is an alias for n.(Directory). If the file isn't directory, a nil value
+// will be returned
+func ToDir(n Node) Directory {
+	d, _ := n.(Directory)
+	return d
+}
+
```

Comment:
FileFrom/DirFrom seem very useful but also *very* magical. While less elegant, have you considered just having multiple functions? (see below)

I'm just worried about dynamic-typing at runtime. That's fine during init/construction (usually) as it can be caught pretty early.